gitmanager.py

```diff
@@ -341,12 +341,43 @@ def merge_pull_request(cls, pr_id, comment=""):
         if not response:
             raise ConnectionError("Cannot connect to GitHub API")
         pr_info = response.json()
-        if pr_info["user"]["login"] != "SmokeDetector":
-            raise ValueError("PR #{} is not created by me, so I can't approve it.".format(pr_id))
-        if "<!-- METASMOKE-BLACKLIST" not in pr_info["body"]:
-            raise ValueError("PR description is malformed. Blame a developer.")
         if pr_info["state"] != "open":
             raise ValueError("PR #{} is not currently open, so I won't merge it.".format(pr_id))
+        if pr_info["user"]["login"] != "SmokeDetector":
+            pr_sha = pr_info["head"]["sha"]
+            response = requests.get("https://api.github.com/repos/{}/pulls/{}/files"
+                                    .format(GlobalVars.bot_repo_slug, pr_id))
+            pr_files = response.json()
+            list_files = ["watched_keywords.txt", "watched_numbers.txt", "bad_keywords.txt",
+                          "blacklisted_numbers.txt", "blacklisted_usernames.txt", "blacklisted_websites.txt"]
+            line_changes = 0
+            for file in range(len(pr_files)):
+                line_changes += int(pr_files[file]["changes"])
+                if pr_files[file]["filename"] not in list_files:
+                    raise ValueError("PR #{} modifies files other than the list files, so I can't approve it."
+                                     .format(pr_id))
+            if line_changes > 6:
+                raise ValueError("PR #{} modifies more than 6 lines, so I can't approve it.".format(pr_id))
+            if not pr_info["mergeable"]:
+                raise ValueError("PR #{} is not mergeable, so I can't approve it.".format(pr_id))
+            github_checks = requests.get("https://api.github.com/repos/{}/pulls/{}/check-runs".
```

Comment:
Perhaps I'm missing something, but this doesn't seem to be a working API. See, e.g., https://api.github.com/repos/Charcoal-SE/SmokeDetector/pulls/5922/check-runs

You probably meant 
```
"https://api.github.com/repos/{}/commits/{}/check-runs"
                                    .format(GlobalVars.bot_repo_slug, pr_sha)
```
(example: https://api.github.com/repos/Charcoal-SE/SmokeDetector/commits/c99d4783e6089651cf3f32099cbb14a1754d2a67/check-runs) 

Also, if you tested this and it is currently getting past this code and allowing merging, it needs more error checking, since an API failure should not allow PRs to be merged.